### ISSUE TITLE
form_rest and model

### ISSUE THREAD

**Issue by jaroslawziolkowski** [kristijanhusak/laravel-form-builder#59]
@kristijanhusak 
When I use form_rest helper fields are not filled data from model. I can't see any option in config to change it.

I need render form without start and close tags because I use more forms class in one view and one request. But to nice formatting impossible is use one class.

**Comment by kristijanhusak** [comment]
@jaroslawziolkowski 
I'm not sure I understand what's the problem? Can you give me some code examples?

If you need to render some form, without opening and closing tags, and to use data from the main form, use child forms.

Behind the scenes when form is created, model binding is done by Laravel's form builder, so if you do not add start and end tags, data will not be bound.

**Comment by jaroslawziolkowski** [comment]
@kristijanhusak 
example form
```
public function buildForm()
    {
        parent::buildForm();
        $this->add('_id','hidden')
            ->add('id','text',['attr'=>['readonly'=>'readonly'],'label'=>trans('common.id')])
            ->add('ean','text',['attr'=>['readonly'=>'readonly'],'label'=>trans('products.ean')])
            ->addTrans('name','text',['label'=>trans('common.name')])
            ->addTrans('description','textarea',['label'=>trans('products.description')])
            ->addTrans('description2','textarea',['label'=>trans('products.description2')])
            ->add('producer','text',['label'=>trans('products.producer')])
            ->addTrans('url','text',['label'=>trans('products.url')])
            ->add('price','text',['attr'=>['readonly'=>'readonly'],'label'=>trans('products.price')])
            ->add('external_code','text',['label'=>trans('products.external_code')])
            ->add('code_max','text',['label'=>trans('products.code_max')])
            ->add('guarantee','text',['label'=>trans('products.guarantee')])
            ->add('approved','checkbox',['default_value'=>1,'label'=>trans('products.approved')])
            ->add('enabled','checkbox',['default_value'=>1,'label'=>trans('products.enabled')])
            ->add('submit','submit',['label'=>trans('common.save')]);
    }
```
controller:
```
$form = $formBuilder->create('\App\Forms\Backend\ProductForm',[
            'url'=>url(\App::getLocale().'/backend/product/card/'.\Hashids::encode_hex($productEan))
        ]);

$model = $dataFromOrm;

$form->setModel($model);
```
view
```
        {!! form_rest($form) !!}
```

Form is correct but not fill data. 
When I replace to
```
    {!! form($form) !!}
```
Fields are fill.

I need use see form without star, close tag but with filled fields from model.

**Comment by kristijanhusak** [comment]
If I understood well, you want to have this form inside some another form, and that's why you do not need start and close tags?

If that's the case, then you could add this form as child to the main form, and pass the model data with main model. Something like this:

```php
class MainForm extends Form {
    public function buildForm()
    {
        $this->add('name', 'text')
               ->add('product', 'form', [
                   'class' => '\App\Forms\Backend\ProductForm'
               ]);
    }
``` 

Controller:
```php
$form = $formBuilder->create('\App\Forms\MainForm',[
            'url'=>url(\App::getLocale().'/backend/product/card/'.\Hashids::encode_hex($productEan))
        ]);
$model = $mainFormData->toArray();
$model['product'] = $dataForProductForm;

$form->setModel($model);
```

view:
```html
{!! form_row($form->product) !!}
```

**Comment by jaroslawziolkowski** [comment]
@kristijanhusak No,
please see my images in last comment.

**Comment by kristijanhusak** [comment]
Try some of these things:

1. call `form_end($form)`, after the `form_rest`.
2. If you do not plan to do anything after the `form_rest`, you can just swap it with `form_end($form)`, and it will automatically call form_rest.

**Comment by jaroslawziolkowski** [comment]
@kristijanhusak 
Thanks, but when I use

```
<div id="tab1">
form_rest($form1)
</div>


<div id="tab2">
form_rest($form2)
</div>


<div id="tab3">
form_rest($form2)
</div>
```

End next 3 times form_end... will be 3 end form tags. Otherwise with data will be only form where I use form_end.

How can I fill data all 3 forms and have correct one close tag?

**Comment by kristijanhusak** [comment]
@jaroslawziolkowski 
`form_rest` is not enough to bind model to a form. 

As i told, you have one main form, which is in your case `Form::open` on the start, and you have 3 forms inside, in tabs. This is how you should do it:

Main form (This one holds all forms)
```php
<?php

class MainForm extends Form
{
    public function buildForm()
    {
        $this
            ->add('tab1form', 'form', [
                'class' => 'App/Forms/Tab1Form'
            ])
            ->add('tab2form', 'form', [
                'class' => 'App/Forms/Tab2Form'
            ])
            ->add('tab3form', 'form', [
                'class' => 'App/Forms/Tab3Form'
            ]);
    }
}
```

Controller:
```php
<?php
public function editAction()
{
    $tab1Model = Tab1::find()->toArray();
    $tab2Model = Tab2::find()->toArray();
    $tab3Model = Tab3::find()->toArray();
    $model = [
        'tab1form' => $tab1Model
        'tab2form' => $tab2Model
        'tab3form' => $tab3Model
    ];

    $form = $formBuilder->create('App\Forms\MainForm', [
        'url' => 'some-url'
    ])->setModel($model);

    return view()->make('view', ['form' => $form]);
}
```

View:
```
{!! form_start($form) !!}
<div id="tab1">
    {!! form_row($form->tab1form) !!}
</div>
<div id="tab2">
    {!! form_row($form->tab2form) !!}
</div>
<div id="tab3">
    {!! form_row($form->tab3form) !!}
</div>
{!! form_end($form) !!}
```

**Comment by jaroslawziolkowski** [comment]
@kristijanhusak 
Thanks,it works. 
Is it possible to remove form name from fields names? Because I use switch in js to change languages fields and arrays will be some difficulty.

**Comment by kristijanhusak** [comment]
@jaroslawziolkowski 
Well, not really. If you really need it that way, then everything needs to be in 1 form, but that is hard to print in views. I believe it's much easier to handle the data after the form submission in the background when it is named. What problem do you have with js? I don't understand what are you switching? It can be probably solved, give me some example.

**Comment by jaroslawziolkowski** [comment]
@kristijanhusak 
Thanks for help. I extended your library and I use addTrans for multi-language fields. Method return 1 text/textarea field and hidden fields with names name__lang_slug

Next in jquery after switch language by user to field name is load vale from hidden field name__lang_slug

I had in this place simple replace and when I added brackets, it sopped working.
But I changed this function to detect array in names and replace inside brackets or only name.

All works well. 

Thank you, as always you are great!
